pandas/core/groupby/groupby.py

```diff
@@ -1120,12 +1117,15 @@ def _resolve_numeric_only(self, numeric_only: bool | lib.NoDefault) -> bool:
             if self.obj.ndim == 2:
                 # i.e. DataFrameGroupBy
                 numeric_only = True
+                obj = self._obj_with_exclusions
+                check = obj._get_numeric_data()
+                if len(obj.columns) and not len(check.columns) and not obj.empty:
+                    warnings.warn("... Explicitly pass numeric_only ...")
```

Comment:
We have alignment on removing this warning - can you do so? A warning will be added in the deprecation of this behavior; that is to be targeted for 1.4 so it is not to be done here.

Reply:
ok so no warning will be added for 1.3.x , only a `FutureWarning` in 1.4